Update docs for nodejs_compat default enable date (#17043)
cloudflare/cloudflare-docs: src/content/docs/workers/runtime-apis/nodejs/index.mdx

## Get Started

> To enable both built-in runtime APIs and polyfills for your Worker, add the [`nodejs_compat`](/workers/configuration/compatibility-dates/#nodejs-compatibility-flag) [compatibility flag](/workers/configuration/compatibility-dates/#nodejs-compatibility-flag) to your `wrangler.toml`, and set your compatibility date to on or after 2024-09-23

Reviewer: ```suggestion
To enable both built-in runtime APIs and polyfills for your Worker, add the [`nodejs_compat`](/workers/configuration/compatibility-dates/#nodejs-compatibility-flag) [compatibility flag](/workers/configuration/compatibility-dates/#nodejs-compatibility-flag) to your `wrangler.toml`, and set your compatibility date to September 23rd, 2024 or later.
```
Per https://developers.cloudflare.com/style-guide/formatting/dates-and-times/

> To enable both built-in runtime APIs and polyfills for your Worker, add the [`nodejs_compat`](/workers/configuration/compatibility-dates/#nodejs-compatibility-flag) [compatibility flag](/workers/configuration/compatibility-dates/#nodejs-compatibility-flag) to your `wrangler.toml`, and set your compatibility date to September 23rd, 2024 or later.

Author: ```suggestion
To enable both built-in runtime APIs and polyfills for your Worker or Pages project, add the [`nodejs_compat`](/workers/configuration/compatibility-dates/#nodejs-compatibility-flag) [compatibility flag](/workers/configuration/compatibility-dates/#nodejs-compatibility-flag) to your `wrangler.toml`, and set your compatibility date to September 23rd, 2024 or later.
```

updated to specify Pages here @mikenomitch

```toml title="wrangler.toml"
compatibility_flags = [ "nodejs_compat" ]
compatibility_date = "2024-09-23"
```

:::note

> Starting on September 23rd, 2024, when your compatibility date is on or after `2024-09-23`, the `nodejs_compat` compatibility flag enables the exact same behavior as the `nodejs_compat_v2` compatibility flag does.

Reviewer: ```suggestion
As of September 23rd, 2024, the `nodejs_compat` compatibility flag enables the exact same behavior as the `nodejs_compat_v2` compatibility flag, as long as your compatibility date is set to September 23rd, 2024 or later.
```
Just trying to make it more active / present tense.

> As of September 23rd, 2024, the `nodejs_compat` compatibility flag enables the exact same behavior as the `nodejs_compat_v2` compatibility flag, as long as your compatibility date is set to September 23rd, 2024 or later.

Reviewer 2: Do you think we should mention that you can move back the compatibility date before 9-23 to get the old behavior? Obviously not something we want to encourage, but if somebody runs into a bug with V2 that's not on V1 then it could give them a workaround.

:::

## Built-in Node.js Runtime APIs

The following APIs from Node.js are provided directly by the Workers Runtime when either `nodejs_compat` or `nodejs_compat_v2` are enabled:

<DirectoryListing />

Unless otherwise specified, implementations of Node.js APIs in Workers are intended to match the implementation in the [Current release of Node.js](https://github.com/nodejs/release#release-schedule).

## Node.js API Polyfills

> When you enable the `nodejs_compat` compatability flag and set your compatibility date to on or after `2024-09-23`, in addition to enabling Node.js APIs in the Workers Runtime, [Wrangler](/workers/wrangler/) will use [unenv](https://github.com/unjs/unenv) to automatically detect uses of Node.js APIs, and add polyfills where relevant.

Reviewer: ```suggestion
When you enable the `nodejs_compat` compatability flag and set your compatibility date to `2024-09-23` or later, in addition to enabling Node.js APIs in the Workers Runtime, [Wrangler](/workers/wrangler/) will use [unenv](https://github.com/unjs/unenv) to automatically detect uses of Node.js APIs, and add polyfills where relevant.
```
Not a strong opinion, but I just think "or later" is a little easier to read.

If an API you wish to use is missing and you want to suggest that Workers support it, please add a post or comment in the
[Node.js APIs discussions category](https://github.com/cloudflare/workerd/discussions/categories/node-js-apis) on GitHub.

## Enable Node.js with Workers

Add the [`nodejs_compat`](/workers/configuration/compatibility-dates/#nodejs-compatibility-flag) [compatibility flag](/workers/configuration/compatibility-dates/#nodejs-compatibility-flag) to your `wrangler.toml`:

```toml title="wrangler.toml"
compatibility_flags = [ "nodejs_compat_v2" ]
```

> ## Enable Node.js in Pages Functions

Reviewer 2: This all still seems relevant. Why drop it?

Author: It works the same way as Workers? Set flag in `wrangler.toml`.
